### PR TITLE
Refactoring OC Mirror deployment to Helm Chart

### PR DESCRIPTION
**Pull request by janboll**
### What this PR does

Jira: 
Link to demo recording:

### CODE REVIEW

**Review by geoberle — APPROVED** [review]

File: image-sync/deployment/Makefile [hunk 1]
```diff
@@ -33,14 +33,34 @@ deploy-component-sync:
...
 	helm uninstall shared-resources --namespace ${NAMESPACE}
 
 undeploy-component-sync:
 	helm uninstall component-sync-cronjob --namespace ${NAMESPACE}
 
-undeploy: undeploy-shared undeploy-component-sync
+undeploy-oc-mirror:
+	helm uninstall oc-mirror-cronjob --namespace ${NAMESPACE}
+
+undeploy: undeploy-shared undeploy-component-sync undeploy-oc-mirror
```
Comment (mociarain) [inline image-sync/deployment/Makefile:64]: General question: "undeploy"... That's a new one for me. Is this a technology specific word or is there any objection to using "remove/delete/uninstall/whatever"? 

Not blocking this PR.

**Review by mociarain — APPROVED** [review]
Small question but LGTM

File: image-sync/deployment/Makefile (hunk 1 above)
Comment (janboll) [reply image-sync/deployment/Makefile:64]: Right, helm uses install/uninstall, but our Makefiles use deploy. I opted for using undeploy cause it's the opposite of deploy and matches our Makefile abstraction.

**Comment by geoberle** [comment]
i was just having a look at the github action job flow. i think the  `build_push_imagesync` and `build_push_ocmirror` steps need to run before `deploy_to_service_cluster` step. can you add them to the `needs` list of `deploy_to_service_cluster`

**Review by geoberle — APPROVED** [review]

**Pull request merged**